api/app/helpers/transcript.ts

```diff
@@ -21,4 +24,44 @@ function generateUniqueHash(content: any) {
   return buffer.toString("base64");
 }
 
-export { generateUniqueStr, generateUniqueHash };
+
+function parseMdToJSON<T extends BaseParsedMdContent>(mdContent: string): T {
+  // This regex is used to match and capture the content between two '---' separators in tstbtc markdown file, typically for front matter, and the rest of the content after the second '---'.
+  const regex = /^---\s*([\s\S]*?)\s*---\s*([\s\S]*)$/;
```

Comment:
Triple dash "---" can be part of the `media` url, for example "https://podcasters.spotify.com/pod/show/chaincode/episodes/Amiti-Uttarwar-and-the-P2P-network---Episode-15-e18v7oq", which breaks this logic.

We can handle that by explicitly looking for `---\n`